File: Domain/Models/POS/CashSession.cs
using System.ComponentModel.DataAnnotations;
using Domain.Enums;

namespace Domain.Models.POS
{
    public class CashSession
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        public Guid CashRegisterId { get; set; }
        public CashRegister? CashRegister { get; set; }

        public Guid CashierUserId { get; set; }

        public DateTime OpenedAt { get; set; } = DateTime.UtcNow;
        public DateTime? ClosedAt { get; set; }

        public decimal OpeningBalance { get; set; }
        public decimal ClosingBalance { get; set; }
        public decimal ExpectedBalance { get; set; }
        public decimal Difference { get; set; }

        public decimal TotalCashSales { get; set; }
        public decimal TotalCardSales { get; set; }
        public decimal TotalOtherSales { get; set; }
        public decimal TotalRefunds { get; set; }

        public CashSessionStatus Status { get; set; } = CashSessionStatus.Open;

        [StringLength(500)]
        public string? Notes { get; set; }

        public ICollection<Sale>? Sales { get; set; }
    }
}

File: Mobile/lib/features/pos/models/cash_session.dart
class CashSession {
  final String id;
  final String cashRegisterId;
  final String? cashRegisterName;
  final String warehouseId;
  final String? warehouseName;
  final String cashierUserId;
  final DateTime openedAt;
  final DateTime? closedAt;
  final double openingBalance;
  final double closingBalance;
  final double expectedBalance;
  final double totalCashSales;
  final double totalCardSales;
  final int status; // 1 = open, 2 = closed

  CashSession({
    required this.id,
    required this.cashRegisterId,
    this.cashRegisterName,
    required this.warehouseId,
    this.warehouseName,
    required this.cashierUserId,
    required this.openedAt,
    this.closedAt,
    required this.openingBalance,
    required this.closingBalance,
    required this.expectedBalance,
    required this.totalCashSales,
    required this.totalCardSales,
    required this.status,
  });

  factory CashSession.fromJson(Map<String, dynamic> json) => CashSession(
        id: json['id'],
        cashRegisterId: json['cashRegisterId'],
        cashRegisterName: json['cashRegisterName'],
        warehouseId: json['warehouseId'] ?? '',
        warehouseName: json['warehouseName'],
        cashierUserId: json['cashierUserId'],
        openedAt: DateTime.parse(json['openedAt']),
        closedAt: json['closedAt'] != null ? DateTime.parse(json['closedAt']) : null,
        openingBalance: (json['openingBalance'] ?? 0).toDouble(),
        closingBalance: (json['closingBalance'] ?? 0).toDouble(),
        expectedBalance: (json['expectedBalance'] ?? 0).toDouble(),
        totalCashSales: (json['totalCashSales'] ?? 0).toDouble(),
        totalCardSales: (json['totalCardSales'] ?? 0).toDouble(),
        status: json['status'] ?? 1,
      );
}

class CashRegister {
  final String id;
  final String name;
  final String code;
  final String warehouseId;
  final String? warehouseName;
  final bool isActive;
  final bool hasOpenSession;

  CashRegister({
    required this.id,
    required this.name,
    required this.code,
    required this.warehouseId,
    this.warehouseName,
    required this.isActive,
    required this.hasOpenSession,
  });

  factory CashRegister.fromJson(Map<String, dynamic> json) => CashRegister(
        id: json['id'],
        name: json['name'] ?? '',
        code: json['code'] ?? '',
        warehouseId: json['warehouseId'],
        warehouseName: json['warehouseName'],
        isActive: json['isActive'] ?? true,
        hasOpenSession: json['hasOpenSession'] ?? false,
      );
}
